test/unit/aggregate_store_test.rb

```diff
@@ -147,6 +147,20 @@ def elasticsearch_store
           assert_equal expected_changes, @instance.aggregate_attribute_changes
         end
 
+        should "return correct changes when an instance is changed more than once" do
+          @instance.name = "First Change"
+          expected_first_changes = {
+            "name" => ["abc", "First Change"]
+          }
+          assert_equal expected_first_changes, @instance.aggregate_attribute_changes
+
+          @instance.name = "Second Change"
+          expected_second_changes = {
+            "name" => ["First Change", "Second Change"]
```

Comment:
Does this match how Rails does it when a record is not saved? That _seems_ surprising to me...since we didn't persist to disk, I would expect `expected_second_changes` to be `["abc", "Second Change"]`

Reply:
Here is what rails does:
```ruby
[2] pry(main)> n.changes
{}
[3] pry(main)> n.name = "hey"
"hey"
[4] pry(main)> n.changes
{"name"=>["Invoca", "hey"]}
[5] pry(main)> n.name = "b"
"b"
[6] pry(main)> n.changes
{"name"=>["Invoca", "b"]}
```

Reply:
Chatted with Alec offline. He is correct that this test is invalid. What we are seeing in web is that even when a transaction has been committed to the database, a second change is still showing the incorrect changes. For example:
```
@org_config = Bing::OrgConfig.create!(approved: true, [....])
@org_config.set!(approved: false)
```
In this case there are two Change Events - the first one for the creation, showing approved going from nil to true. The second one is showing that approved is going from false to false.

We chatted about how Aggregate is [aware of transaction commits](https://github.com/Invoca/aggregate/blob/7f1521783a25429eef05a9075b38cd973a151c56/lib/aggregate/container.rb#L19), and that we should make the change over there instead of where I made it.

Update coming for that soon.